plutus-contract/test/Spec/Contract/Tx/Constraints/RequiredSigner.hs

```diff
@@ -156,3 +169,98 @@ mustBeSignedByTypedValidator = Scripts.mkTypedValidator @UnitTest
+
+    utxos <- ownUtxos
+    myAddr <- ownAddress
+    let get3 (a:b:c:_) = (a, b, c)
+        get3 _         = error "Spec.Contract.TxConstraints.get3: not enough inputs"
+        ((utxoRef, utxo), (utxoRefForBalance1, _), (utxoRefForBalance2, _)) = get3 $ M.toList utxos
+        vh = fromJust $ Ledger.toValidatorHash mustReferenceOutputV2ValidatorAddress
+        lookups1 = Constraints.unspentOutputs utxos
+               <> Constraints.plutusV2OtherScript mustReferenceOutputValidatorV2
+               <> Constraints.paymentPubKey pk
+        tx1 = Constraints.mustPayToOtherScriptWithDatumInTx
+                vh
+                (Ledger.Datum $ PlutusTx.toBuiltinData utxoRef)
+                (Ada.adaValueOf 5)
+          <> Constraints.mustSpendPubKeyOutput utxoRefForBalance1
+          <> Constraints.mustUseOutputAsCollateral utxoRefForBalance1
+          <> Constraints.mustPayToAddressWithReferenceValidator
+                myAddr
+                vh
+                Nothing
+                (Ada.adaValueOf 30)
+    submitTxConfirmed $ mkTx lookups1 tx1
+
+    -- Trying to unlock the Ada in the script address
+    scriptUtxos <- utxosAt mustReferenceOutputV2ValidatorAddress
+    utxos' <- ownUtxos
+    let
+        scriptUtxo = fst . head . M.toList $ scriptUtxos
+        refScriptUtxo = head . M.keys . M.filter (has $ Tx.decoratedTxOutReferenceScript . _Just) $ utxos'
+        lookups2 = Constraints.unspentOutputs (M.singleton utxoRef utxo <> scriptUtxos <> utxos')
+        tx2 = Constraints.mustReferenceOutput utxoRef
+          <> Constraints.mustSpendScriptOutputWithReference scriptUtxo Ledger.unitRedeemer refScriptUtxo
+          <> Constraints.mustSpendPubKeyOutput utxoRefForBalance2
+          <> Constraints.mustUseOutputAsCollateral utxoRefForBalance2
+          <> Constraints.mustBeSignedBy pkh
+    logInfo @String $ "Required Signatories: " ++ show (Constraints.requiredSignatories tx2)
+    submitTxConfirmed $ mkTx lookups2 tx2
+
+cardanoTxOwnWallet :: TestTree
+cardanoTxOwnWallet =
+    let pk  = mockWalletPaymentPubKey     w1
+        pkh = mockWalletPaymentPubKeyHash w1
+        trace = do
+            void $ Trace.activateContractWallet w1 $ cardanoTxOwnWalletContract pk pkh
+            void Trace.nextSlot
+    in checkPredicateOptions
+        (changeInitialWalletValue w1 (const $ Ada.adaValueOf 1000) defaultCheckOptions)
+        "own wallet's signature passes on-chain mustBeSignedBy validation with cardano tx" (assertValidatedTransactionCount 2) (void trace)
+
+cardanoTxOtherWalletNoSigningProcess :: TestTree
+cardanoTxOtherWalletNoSigningProcess =
+    let pk  = mockWalletPaymentPubKey     w2
+        pkh = mockWalletPaymentPubKeyHash w2
+        trace = do
+            void $ Trace.activateContractWallet w1 $ cardanoTxOwnWalletContract pk pkh
+            void Trace.nextSlot
+    in checkPredicateOptions
+        (changeInitialWalletValue w1 (const $ Ada.adaValueOf 1000) defaultCheckOptions)
```

Comment:
why is lower wallet balance important for these tests?

Reply:
We don't have auto-balancing for CardanoTx yet, so fixing the wallet balance is needed to balance it (I could have done the correct computation to avoid it but I don't think it worths the effort)

Reply:
ok, would you mind mentioning it inline so we can take it out when there is balancing? thanks